### PR TITLE
Bump LLD to get dsymutil and run it through the pkgimgs

### PR DESCRIPTION
**Pull request by gbaraldi**
Finishes fixing https://github.com/JuliaLang/julia/issues/48020

### CODE REVIEW

**Comment by vchuravy** [comment]
I am afk right now, but for the LRU cache of cachefiles we count the number of files in the directory (and also rn then accordingly) you will need to add dsym there

**Comment by gbaraldi** [comment]
Where do we manage it? In Pkg itself or is it in julia?

**Comment by vchuravy** [comment]
Looks like I did the filtering on `.ji` already, but you will have to delete the dsym here https://github.com/JuliaLang/julia/blob/3db0cc20eba14978c45cd91f05a9b89b1dc0e38a/base/loading.jl#L2227

File: base/loading.jl [hunk 1]
```diff
@@ -2224,6 +2224,7 @@ function compilecache(pkg::PkgId, path::String, internal_stderr::IO = stderr, in
                     rm(evicted_cachefile; force=true)
                     try
                         rm(ocachefile_from_cachefile(evicted_cachefile); force=true)
+                        rm(ocachefile_from_cachefile(evicted_cachefile) * ".dSYM"; force=true)
```
[the remark this answers is not in the document]
Comment (vtjnash) [reply base/loading.jl:2227]: Needs `recursive=true` also

[the remark this answers is not in the document]
Comment (vchuravy) [reply base/loading.jl]: Why is `recursive=true` necessary? Is `dsym` a directory?

[the remark this answers is not in the document]
Comment (gbaraldi) [reply base/loading.jl]: Yes

**Review by vchuravy — APPROVED** [review]

**Pull request merged**